addons/talo/apis/leaderboards_api.gd

```diff
@@ -1,77 +1,97 @@
-class_name LeaderboardsAPI extends TaloAPI
+## LeaderboardsAPI
+extends TaloAPI
 ## An interface for communicating with the Talo Leaderboards API.
 ##
 ## This API is used to read and update leaderboards in your game. Leaderboards are used to track player scores and rankings.
 ##
 ## @tutorial: https://docs.trytalo.com/docs/godot/leaderboards
 
-var _entries_manager = TaloLeaderboardEntriesManager.new()
+var _entries_manager := TaloLeaderboardEntriesManager.new()
 
 ## Get a list of all the entries that have been previously fetched or created for a leaderboard.
 func get_cached_entries(internal_name: String) -> Array:
 	return _entries_manager.get_entries(internal_name)
 
 ## Get a list of all the entries that have been previously fetched or created for a leaderboard for the current player.
-func get_cached_entries_for_current_player(internal_name: String) -> Array:
+func get_cached_entries_for_current_player(internal_name: String) -> Array[TaloLeaderboardEntry]:
 	if Talo.identity_check() != OK:
 		return []
 
 	return _entries_manager.get_entries(internal_name).filter(
-		func (entry: TaloLeaderboardEntry):
+		func(entry: TaloLeaderboardEntry) -> bool:
 			return entry.player_alias.id == Talo.current_alias.id
 	)
 
 ## Get a list of entries for a leaderboard. The page parameter is used for pagination.
-func get_entries(internal_name: String, page: int, alias_id = -1, include_archived = false) -> Array:
-	var url = "/%s/entries?page=%s"
-	var url_data = [internal_name, page]
+func get_entries(internal_name: String, page: int, alias_id := -1, include_archived := false) -> EntriesPage:
+	var url := "/%s/entries?page=%s"
+	var url_data := [internal_name, page]
 
 	if alias_id != -1:
 		url += "&aliasId=%s"
-		url_data += alias_id
+		url_data.push_back(alias_id)
 
 	if include_archived:
 		url += "&withDeleted=1"
 
-	var res = await client.make_request(HTTPClient.METHOD_GET, url % url_data)
+	var res := await client.make_request(HTTPClient.METHOD_GET, url % url_data)
 
 	match (res.status):
 		200:
-			var entries: Array = res.body.entries.map(
-				func (data: Dictionary):
-					var entry = TaloLeaderboardEntry.new(data)
+			var entries: Array = Array(res.body.entries.map(
+				func(data: Dictionary) -> TaloLeaderboardEntry:
+					var entry := TaloLeaderboardEntry.new(data)
 					_entries_manager.upsert_entry(internal_name, entry)
 
 					return entry
-			)
-			return [entries, res.body.count, res.body.isLastPage]
+			), TYPE_OBJECT, (TaloLeaderboardEntry as Script).get_instance_base_type(), TaloLeaderboardEntry)
```

Comment:
I'm a little confused about this syntax - could you point me to some documentation on it?

Reply:
A typed array constructor. You can refer in editor help.
`Array Array(base: Array, type: int, class_name: StringName, script: Variant)`.

Reply:
ah great stuff, didn't know this existed!